cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -175,17 +175,18 @@ private ProgramStatus getExistingAppProgramStatus(ApplicationSpecification appSp
         }
         ensureAccess(programId);
 
-        if ((programId.getType() == ProgramType.MAPREDUCE || programId.getType() == ProgramType.SPARK) &&
-          !store.getRuns(programId, ProgramRunStatus.RUNNING, 0, Long.MAX_VALUE, 1).isEmpty()) {
-          // MapReduce program exists and running as a part of Workflow
-          return ProgramStatus.RUNNING;
-        }
-        return ProgramStatus.STOPPED;
+        // If no runtime info exists for a program, the program status can be determined based on if the store has
+        // a RUNNING run record or not
+        return (store.getRuns(programId, ProgramRunStatus.RUNNING, 0, Long.MAX_VALUE, 1).isEmpty()) ?
```

Comment:
No it's not. This logic exists because the `runtimeInfo` does not exist, but the program does exist. This means that the program is either in a workflow, or has finished. 

Checking the store for a running run record tells us then if it the program's final status has been persisted.